architecture/ambient/ztunnel.md

```diff
@@ -0,0 +1,236 @@
+# Ztunnel
+
+This document provides an overview of the architecture and design decisions around Ztunnel, the node-proxy component in ambient mode.
+
+## Background and motivation
+
+Motivations to implement ztunnel generally came from two areas.
+
+First, and most importantly, it serves as a means to implement the real goal: waypoints.
+For various reasons outside the scope of this document, there is a desire to move from a sidecar based architecture to a "remote proxy" architecture.
+However, this has one glaring issue: how do we get the traffic to the remote proxies, while maintaining the zero-trust properties that Istio is built upon?
+
+A secondary goal was to enable was to enable a smoother on-ramp from "Zero" to "Getting some value".
+Historically, Istio had to really be consumed all-or-nothing for things to work as expected.
+In particular, an easy answer to "I just want to get have mTLS everywhere, then I can think about adopting the rest of service mesh" was desired.
+
+## Goals
+
+Ztunnel should:
+* **Not break users**. This means that deploying Ztunnel should retain all existing Kubernetes behavior.
+  * This includes UDP, non-compliant HTTP, server-first protocols, stateful sets, external services, etc.
+  * Explicitly opting into behavioral changes can be acceptable. For example, introducing Istio multi-cluster semantics.
+* Ensure traffic between mesh workloads is securely encrypted with an Istio identity.
+* Be lightweight enough to not limit adoption.
+  * This puts a much tighter budget on CPU, memory, latency, and throughput requirements than traditional Istio.
```

Comment:
I think it might be clearer to end the sentence with "than traditional Istio **sidecars**".